image_processing/src/views/layout_utils.py

```diff
@@ -0,0 +1,548 @@
+                          tooltip='Click to generate Histogram')
+            ]
+        )
+
+    ]
+    image_filename = ip.plot_image(q.client.image)
+    content, = await q.site.upload([image_filename])
+    os.remove(image_filename)
+    q.page['original_image'].content = f'''![plot]({content})'''
+    q.page['transformed_image'].content = f'''Your histogram will be displayed here'''
+    await q.page.save()
+
+
+async def rgb2gray_layout(q: Q):
+    if q.client.blur_controls:
+        await remove_blur_controls(q)
+        q.client.blur_control = False
+    if q.client.histogram_match:
+        await remove_histogram_match(q)
+        q.client.histogram_match = False
+
+    q.page['controls'].items = [
+        ui.buttons(
+            items=[
+                ui.button(name='rgb2gray', label='Change RGB to GrayScale', primary=True,
+                          tooltip='Click to apply RGB2GRAY'),
+                ui.button(name='reset_rgb2gray', label='Reset Image', tooltip='Click to Reset')
+            ]
+        )
+
+    ]
+    image_filename = ip.plot_image(q.client.image)
+    content, = await q.site.upload([image_filename])
+    os.remove(image_filename)
+    q.page['original_image'].content = f'''![plot]({content})'''
+    q.page['transformed_image'].content = f'''![plot]({content})'''
+    await q.page.save()
+
+
+async def rotation_layout(q: Q):
+    if q.client.blur_controls:
+        await remove_blur_controls(q)
+        q.client.blur_control = False
+    if q.client.histogram_match:
+        await remove_histogram_match(q)
+        q.client.histogram_match = False
+
+    q.page['controls'].items = [
+        ui.slider(name='rotate', label='Translation on X axis', min=-180, max=180, step=1),
+        ui.textbox(name='x_point', label='X coordinate to rotate'),
+        ui.textbox(name='y_point', label='Y coordinate to rotate'),
+        ui.buttons(
+            items=[
+                ui.button(name='rotation', label='Apply Rotation', primary=True,
+                          tooltip='Click to apply rotation transformation'),
+                ui.button(name='reset_rotation', label='Reset Rotation', tooltip='Click to Reset')
+            ]
+        )
+
+    ]
+    image_filename = ip.plot_image(q.client.image)
+    content, = await q.site.upload([image_filename])
+    os.remove(image_filename)
+    q.page['original_image'].content = f'''![plot]({content})'''
+    q.page['transformed_image'].content = f'''![plot]({content})\n
+            Last Rotation = {q.args.rotation} '''
+    await q.page.save()
+
+
+async def translation_layout(q: Q):
+    if q.client.blur_controls:
+        await remove_blur_controls(q)
+        q.client.blur_control = False
+    if q.client.histogram_match:
+        await remove_histogram_match(q)
+        q.client.histogram_match = False
+
+    q.page['controls'].items = [
+        ui.slider(name='translate_X', label='Translation on X axis', min=0, max=360, step=1),
```

Comment:
Can we document why 360 is used here, or could this be dynamic per image? I see that's larger than the demo image (~215) so it's possible to do translations on the Y axis that completely remove the image.